src/datepicker/calendar.js

```diff
@@ -285,13 +269,50 @@ export default class Calendar extends React.Component<
   onDayMouseOver = (data: {event: Event, date: Date}) => {
     const {date} = data;
     this.setState({highlightedDate: date});
-    this.props.onDayMouseOver(data);
+    this.props.onDayMouseOver && this.props.onDayMouseOver(data);
   };
 
   onDayMouseLeave = (data: {event: Event, date: Date}) => {
     const {date} = data;
     this.setHighlightedDate(date);
-    this.props.onDayMouseLeave(data);
+    this.props.onDayMouseLeave && this.props.onDayMouseLeave(data);
+  };
+
+  handleDateChange = (data: {date: ?Date | Array<Date>}) => {
```

Comment:
What do you think about having the date value always as an array (range or single datepickjer). Since we're releasing a major update we could add this change too. It will align it with the Selects API approach, simplify the internals, and remove the confusion regarding passing an initial empty array value for the range picker.

Reply:
i'd rather enable a better developer experience and contain the complexity for ourselves